src/components/TabsCard.tsx

```diff
@@ -111,63 +111,116 @@ const TabsCard = ({
+
+//   const res = [];
+//   let tempList = [];
+//   for (let i = 0; i < arr.length; i++) {
+//     if (i % 3 !== 0) tempList.push(arr[i]);
+//     else {
+//       res.push(tempList);
+//       tempList = [];
+//     }
+//   }
+
+//   // console.log(diseaseInfo.replaceAll('[', '').replaceAll(']', '').split(','));
+//   console.log(res);
+//   // .map((val) => {
+//   //   return <p key={'yessir'}>{val}</p>;
+//   // })
+//   // .join('\n\n');
+//   res.splice(0, 1);
+//   return res.map(([one, two, three]) => {
+//     console.log([one, two, three]);
+//     return (
+//       <>
+//         <div>
+//           {/* {one}
+//           <br />
+//           {two}
+//           <br />
+//           {three}
+//           <br /> */}
+//           <a href={two}>{one}</a>
+//         </div>
+//       </>
+//     );
+//   });
+// }
+
 function parseDiseaseInfo(diseaseInfo: string) {
-  const list = [];
   let temp = '';
+  const list = [];
 
   for (let i = 0; i < diseaseInfo.length; i++) {
-    if (diseaseInfo[i] === '[') {
-      continue;
-    } else if (diseaseInfo[i] === ']') {
+    if (diseaseInfo[i] === '[') continue;
+    else if (diseaseInfo[i] === ']') {
       list.push(temp);
-      console.log('pushing ' + temp);
-      temp = '';
-      continue;
-    } else if (diseaseInfo[i] === ',') {
-      if (temp !== '') list.push(temp);
+      console.log('pushing ' + temp.trim());
       temp = '';
       continue;
     }
     temp += diseaseInfo[i];
   }
 
-  console.log('hi');
-  console.log(list.join('\n'));
-  const arr = diseaseInfo.replaceAll('[', '').replaceAll(']', '').split(',');
-
-  const res = [];
-  let tempList = [];
-  for (let i = 0; i < arr.length; i++) {
-    if (i % 3 !== 0) tempList.push(arr[i]);
-    else {
-      res.push(tempList);
-      tempList = [];
+  const diseaseList = [];
+  for (let i = 0; i < list.length; i++) {
+    const subList = list[i].trim().split(', ');
+    for (let j = 0; j < subList.length; j++) {
+      if (subList[j] === '') {
+        console.log('got here');
+        subList.splice(j, 1);
+        console.log(subList);
+      }
+    }
+
+    if (subList.length >= 3) {
+      diseaseList.push(subList);
     }
   }
 
-  // console.log(diseaseInfo.replaceAll('[', '').replaceAll(']', '').split(','));
-  console.log(res);
-  // .map((val) => {
-  //   return <p key={'yessir'}>{val}</p>;
-  // })
-  // .join('\n\n');
-  res.splice(0, 1);
-  return res.map(([one, two, three]) => {
-    console.log([one, two, three]);
-    return (
-      <>
-        <div>
-          {/* {one}
-          <br />
-          {two}
-          <br />
-          {three}
-          <br /> */}
-          <a href={two}>{one}</a>
-        </div>
-      </>
-    );
-  });
+  console.log(diseaseList);
+
+  return (
+    <div>
+      {diseaseList.map(([id, ...rest]) => {
+        return (
+          <div key={id}>
+            <p>
+              {id}
+              <br />
+              {rest.slice(0, rest.length - 1)}
```

Comment:
add separator between individual words.
might have to write a function for this